pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -71,42 +73,72 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 	needReboot = false
 	err = nil
 	attributesToChange = map[string]mlnxNic{}
+	mellanoxNicsStatus = map[string]map[string]*sriovnetworkv1.InterfaceExt{}
+	mellanoxNicsSpec = map[string]*sriovnetworkv1.Interface{}
+	processedNics := map[string]bool{}
 
+	for _, iface := range new.Status.Interfaces {
```

Comment:
The PF information would not change during runtime. So maybe you can consider to use a package scoped variable, and initialize it once.

Reply:
changed